> pandas/core/arrays/sparse/array.py:886: error: Non-overlapping identity check (left operand type: "Union[Union[Union[int, integer[Any]], Union[slice, List[int], ndarray[Any, Any]]], Tuple[Union[int, ellipsis], ...]]", right operand type: "ellipsis")  [comparison-overlap]


otherwise lgtm.

--- a/pandas/core/arrays/sparse/array.py
+++ b/pandas/core/arrays/sparse/array.py
@@ -883,7 +883,11 @@ class SparseArray(OpsMixin, PandasObject, ExtensionArray):
 
         if isinstance(key, tuple):
             key = unpack_tuple_and_ellipses(key)
-            if key is Ellipsis:
+            # Non-overlapping identity check (left operand type:
+            # "Union[Union[Union[int, integer[Any]], Union[slice, List[int],
+            # ndarray[Any, Any]]], Tuple[Union[int, ellipsis], ...]]",
+            # right operand type: "ellipsis")
+            if key is Ellipsis:  # type: ignore[comparison-overlap]
                 raise ValueError("Cannot slice with Ellipsis")
 
         if is_integer(key):

